### PR TITLE
Switching webserver from thin to puma for consistency across alphasights

### PR DESCRIPTION
**Pull request by delkopiso**
## What this does
- Removes thin webserver.
- Adds puma.
- Adds puma config as recommended by Heroku.

### Pre-deploy tasks
- [x] Set `ENV['MAX_THREADS']=1` and `ENV['MIN_THREADS']=1`
- [x] Deploy to staging and confirm all is well.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,18 +1,17 @@
 ruby '2.3.0'
 
 source 'https://rubygems.org'
-source 'https://rails-assets.org'
```
Comment (carlosepp) [inline Gemfile]: don´t we need this?

Comment (delkopiso) [inline Gemfile]: I don't think so since this is just an API and we are not really serving any assets.

Comment (carlosepp) [inline Gemfile]: :ok_hand:

**Comment by carlosepp** [comment]
:+1:

**Pull request merged**